test/SILOptimizer/definite-init-wrongscope.swift

```diff
@@ -0,0 +1,43 @@
+// RUN: %target-swift-frontend -primary-file %s -Onone -emit-sil -Xllvm \
+// RUN:   -sil-print-after=definite-init -Xllvm \
+// RUN:   -sil-print-only-functions=$S3del1MC4fromAcA12WithDelegate_p_tKcfc \
+// RUN:   -Xllvm -sil-print-debuginfo -o /dev/null 2>&1 | %FileCheck %s
+
+public protocol DelegateA {}
+public protocol DelegateB {}
+public protocol WithDelegate
+{
+    var delegate: DelegateA? { get }
+    func f() throws -> Int
+}
+public enum Err: Swift.Error {
+    case s(Int)
+}
+public class C {}
+public class M {
+    let field: C
+    var value : Int
+    public init(from d: WithDelegate) throws {
+        guard let delegate = d.delegate as? DelegateB
+        else { throw Err.s(0) }
+        self.field = C()
+        let i: Int = try d.f()
+        value = i
+    }
+}
+
+// Make sure the expanded sequence gets the right scope.
+
+// CHECK:   %49 = begin_access [modify] [dynamic] %48 : $*C, loc{{.*}}:23:20, scope 2
```

Comment:
`%[[BEGIN_ACCESS:[0-9]+]] = begin_access ...` would be more resilient

Reply:
Cool, I'm going to change it as a follow up